api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -82,10 +78,6 @@ func (mw *azureManagedMachinePoolWebhook) Default(ctx context.Context, obj runti
 		m.Spec.OSType = ptr.To(DefaultOSType)
 	}
 
-	if ptr.Deref(m.Spec.ScaleSetPriority, "") == string(armcontainerservice.ScaleSetPrioritySpot) && m.Spec.SpotMaxPrice == nil {
-		m.Spec.SpotMaxPrice = ptr.To(resource.MustParse("-1"))
-	}
```

Comment:
Do we no longer want to set the default max spot price?

Reply:
I don't think this provides any benefit since #4112 doesn't manifest without this anymore.